zerver/tests/test_subs.py

```diff
@@ -634,13 +799,21 @@ def test_try_make_stream_public_with_private_history(self) -> None:
             "is_private": orjson.dumps(False).decode(),
             "history_public_to_subscribers": orjson.dumps(False).decode(),
         }
-        stream_id = get_stream("public_stream", realm).id
+        stream_id = self.subscribe(user_profile, "public_stream").id
         result = self.client_patch(f"/json/streams/{stream_id}", params)
         self.assert_json_success(result)
         stream = get_stream("public_stream", realm)
         self.assertFalse(stream.invite_only)
         self.assertTrue(stream.history_public_to_subscribers)
 
+        messages = get_topic_messages(user_profile, stream, "stream events")
+        self.assert_length(messages, 1)
+        expected_notification = (
+            f"@_**King Hamlet|{user_profile.id}** changed the [access permissions](/help/stream-permissions) "
+            "for this stream from **Public** to **Public**."
```

Comment:
Why is this one public to public?  This seems like a bug. 

(If it's not, please add a comment explaining why we're expecting this surprising notification).

Reply:
This is not a bug, will add an explanatory comment. Thank you! :)